tianshou/data/buffer.py

```diff
@@ -394,6 +480,17 @@ def reset(self) -> None:
             if isinstance(self._meta.__dict__[k], list):
                 self._meta.__dict__[k] = []
 
+    def __getstate__(self) -> dict:
+        return self.__dict__
+
+    def __setstate__(self, state: Dict[str, Any]) -> None:
+        """Unpickling interface.
+
+        We need it because pickling buffer does not work out-of-the-box
+        ("buffer.__getattr__" is customized).
+        """
+        self.__dict__.update(state)
+
```

Comment:
I don't think it is necessary since it is equal to the implementation in `ReplayBuffer`. It will directly inherit that function. Same as the below.